pkg/kubectl/cmd/rollout/rollout_pause.go

```diff
@@ -89,32 +90,44 @@ func (o *PauseConfig) CompletePause(f *cmdutil.Factory, cmd *cobra.Command, out
 		return err
 	}
 
-	infos, err := resource.NewBuilder(o.Mapper, o.Typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+	r := resource.NewBuilder(o.Mapper, o.Typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
 		NamespaceParam(cmdNamespace).DefaultNamespace().
 		FilenameParam(enforceNamespace, o.Recursive, o.Filenames...).
 		ResourceTypeOrNameArgs(true, args...).
-		SingleResourceType().
+		ContinueOnError().
 		Latest().
-		Do().Infos()
+		Flatten().
+		Do()
+	err = r.Err()
 	if err != nil {
 		return err
 	}
-	if len(infos) != 1 {
-		return fmt.Errorf("rollout pause is only supported on individual resources - %d resources were found", len(infos))
+
+	err = r.Visit(func(info *resource.Info, err error) error {
```

Comment:
Correct, its to gather information. Using `Infos()` would return an error early before all resources have been processed, rather than aggregating them and showing them at the end.

By using `Visit()` we not only traverse the resources, but we also aggregate errors encountered. This ensures that the resources that can be visited will be processed even if an error were to arise on another resource.